### ISSUE TITLE
Add another PVS function "AddEntityToPVS"

### ISSUE THREAD

**Issue by HellaMadMax** [Facepunch/garrysmod-requests#245]
There is currently "AddOriginToPVS" but there are times where we only want that specific entity to be added, and having a function to only add an entity to pvs would be great.

**Comment by mcd1992** [comment]
This issue isn't really related to the PVS as the PVS is just the source engine's way of finding out which entities are visible to a client and need to be networked. Adding a PVS to the players PVS does give you the desired result but it's networking everything else because that's just how the PVS works.

I haven't read into this too much, nor have I tested anything I'm saying below, but from what I'm seeing we currently have a binding to ENTITY:AddEFlags which will call ENTITY:UpdateTransmitState. As a temporary workaround I think you can re-define the entity's UpdateTransmitState function to TRANSMIT_ALWAYS then add the EFL_FORCE_CHECK_TRANSMIT flag to the entity with AddEFlags. I think this will network the entity to ALL players, not sure how to go about networking an entity to a single/group of players.

If I had to guess I think what you want is a hook like [ENTITY:ShouldTransmit](https://github.com/ValveSoftware/source-sdk-2013/blob/master/mp/src/game/server/baseentity.h#L499) which would allow us to change the CCheckTransmitInfo's m_pClientEnt to specific clients. I'll have to read up some more when I get home from work on how the engine picks which players to network entities to.

**Comment by LeQuackers** [comment]
I agree with HellaMadMax and Matt's comments about how this would be beneficial. I think a stealth system that networks a person to nearby players for a few seconds if they made a loud enough noise would be an interesting concept. AddEntityToPVS as well as ENTITY:ShouldTransmit have lots of potential.

**Comment by sammyt291** [comment]
Id love to see this added also,
It would fit perfectly for what im trying to find a way around: [InfoHUD](https://facepunch.com/showthread.php?t=1551100)

Having a way to set always transmit AFTER an ent is created would have great potential.

**Comment by chelog** [comment]
I would definitely love to see `AddEntityToPVS( ent )` function to use inside `GM:SetupPlayerVisibility( ply, viewEnt )` just like [AddOriginToPVS](https://wiki.garrysmod.com/page/Global/AddOriginToPVS).
It would be great for spectators, some clientside gui maps and lots of different stuff. It won't allow hackers to get position when they want, it's up to server to decide whether to include entity in player's PVS or not so no benefits for hackers here.

**Comment by nkilmer** [comment]
I'd also find an AddEntityToPVS function useful. I'm trying to create a TTT add-on that allows the Detective to track a single player wherever they go and the lack of a good way to force player positions to always be transmitted is making it quite painful. I have a hacky workaround but AddEntityToPVS would be the 'correct' way to do what I'm trying to accomplish.

**Comment by blobles-dev** [comment]
> I'd also find an AddEntityToPVS function useful. I'm trying to create a TTT add-on that allows the Detective to track a single player wherever they go and the lack of a good way to force player positions to always be transmitted is making it quite painful. I have a hacky workaround but AddEntityToPVS would be the 'correct' way to do what I'm trying to accomplish.

@robotboy655 any chance of this? Ive had to do some hacky non performant stuff to achieve this, would be great

**Comment by robotboy655** [comment]
Would this just be basically duplicate of https://github.com/Facepunch/garrysmod-requests/issues/2410 ? (or rather backwards, but still)

**Comment by wrefgtzweve** [comment]
They'd achieve the same i think yeah

**Comment by MrGenie151** [comment]
Needing this for an assym gamemode with a monster with good hearing
